test/python/transpiler/test_linear_functions_passes.py

```diff
@@ -20,7 +20,7 @@
 from qiskit.circuit import QuantumCircuit, Qubit, Clbit
 from qiskit.transpiler.passes.optimization import CollectLinearFunctions
 from qiskit.transpiler.passes.synthesis import (
-    LinearFunctionsSynthesis,
+    HighLevelSynthesis,
```

Comment:
We should maintain some tests of the old pass to ensure it works as is during the deprecation period.

Reply:
I have added a test that the old functionality still works and a bug fix (shame on me, I did break the functionality), see [0309abe](https://github.com/Qiskit/qiskit-terra/pull/9041/commits/0309abe5da0716ad1107df355992c7ccefe48e9a). I believe that a single test suffices, since most of the tests actually check the orhogonal block collection aspect of resynthesizing linear functions.

There was also a suggestion to extend the `@combine(do_commutative_analysis=[False, True])` to something like`@combine(do_commutative_analysis=[False, True], synthesis_pass=[HighLevelSynthesis, LinearFunctionsSynthesis])`, however I did not end up using this since in one case we want to call 

```synthesized_circuit = PassManager(synthesis_pass()).run(optimized_circuit)```

and in the other case we want to call 

```
with self.assertWarns(DeprecationWarning):
    synthesized_circuit = PassManager(synthesis_pass()).run(optimized_circuit)
```

and I did not think of a clever way to nicely combine the two.